todo-app/src/App.js

```diff
@@ -1,13 +1,83 @@
-import React, { Component } from 'react';
+//jshint esnext: true
+
+import React from 'react';
+
+import TodoList from './Components/todoList';
+import AddTodoItem from './Components/AddTodoItem';
+import './App.css';
+
+
+export default class App extends React.Component {
+
+  
+  state= {
+    todos: [],
+    isEditing: null
+  }
+
+
+  handleToggleEditing = (id) => {
+    this.setState({ isEditing: id || null })
+  }
+
+  handleAddTodo = (text, deadline) => {
+    
+    const newItem = {
+        id: this.state.todos.length + 1 || 1,
+        text,
+        deadline,
+        done: false
+      }
+      const oldTodos = this.state.todos || [];
+      
+      this.setState({ todos: [...oldTodos, newItem]})
+    }
+
+  handleToggleDone = (id) => {
+    const newTodos = [...this.state.todos].map(item => {
+      if (item.id === id){
+        item.done = !item.done; 
```

Comment:
This is something from last week, but try to fix it here: in this line, you're *mutating* (changing) an existing todo item. To make this function not mutate, return a new item with the `done` flag inversed. If you're not changing the item (i.e. if `item.id !== id`), you can return the original item.